src/main/java/seedu/address/model/pet/Appointment.java

```diff
@@ -48,16 +57,25 @@ public Appointment(String value) {
         if (value.equals("")) {
             this.value = value;
             this.location = null;
+            this.date = null;
+            this.time = null;
             this.dateTime = null;
         } else {
             String[] appointmentDetails = value.split(" at ");
             String retrievedDateTime = appointmentDetails[0];
             String retrievedLocation = appointmentDetails[1];
-            DateTimeFormatter formatOfRetrievedDateTime = DateTimeFormatter.ofPattern("MMM-dd-yyyy h:mm a")
+            String retrievedDate = retrievedDateTime.split(" ", 2)[0].trim();
+            String retrievedTime = retrievedDateTime.split(" ", 2)[1].trim();
+
+            DateTimeFormatter formatOfRetrievedDate = DateTimeFormatter.ofPattern("MMM-dd-yyyy")
+                .withLocale(Locale.ENGLISH);
+            DateTimeFormatter formatOfRetrievedTime = DateTimeFormatter.ofPattern("h:mm a")
                 .withLocale(Locale.ENGLISH);
             this.value = value;
             this.location = retrievedLocation;
-            this.dateTime = LocalDateTime.parse(retrievedDateTime, formatOfRetrievedDateTime);
+            this.date = LocalDate.parse(retrievedDate, formatOfRetrievedDate);
+            this.time = LocalTime.parse(retrievedTime, formatOfRetrievedTime);
+            this.dateTime = date.atTime(time);
```

Comment:
Since we are already splitting the date and time into separate fields, is there a particular reason for keeping this dateTime field within Appointment?